src/XML/src/FMI3/fmi3_xml_model_structure.c

```diff
@@ -454,14 +481,56 @@ int fmi3_xml_handle_ContinuousStateDerivative(fmi3_xml_parser_context_t *context
     return 0;
 }
 
-// TODO
 int fmi3_xml_handle_ClockedState(fmi3_xml_parser_context_t *context, const char* data) {
     if(!data) {
         fmi3_xml_model_description_t* md = context->modelDescription;
         fmi3_xml_model_structure_t* ms = md->modelStructure;
 
-        // return fmi3_xml_parse_unknown(context, fmi3_xml_elmID_ClockedState, &ms->ClockedStates, ms->clockedStateDeps);
-        return -1; // TODO
+        /* perform the parsing */
+        if (fmi3_xml_parse_unknown(context, fmi3_xml_elmID_ClockedState,
+                                   &ms->clockedStates, ms->clockedStateDeps)) {
+            return -1;
+        }
+
+        /* validate return values */
+        fmi3_xml_variable_t* clock_var = (fmi3_xml_variable_t*)jm_vector_get_last(jm_voidp)(&ms->clockedStates);
+
+        /* clock attribute required */
+        if (!fmi3_xml_is_clocked(clock_var)){
```

Comment:
```suggestion
        if (!fmi3_xml_is_clocked(clock_var)) {
```
On a bunch of places